src/StackExchange.Metrics/Handlers/StatsdMetricHandler.cs

```diff
@@ -13,10 +13,104 @@
+        /// Constructs a new <see cref="StatsdMetricHandler" /> pointing at the specified host and port.
+        /// </summary>
+        /// <param name="host">
+        /// Host of a statsd endpoint.
+        /// </param>
+        /// <param name="port">
+        /// Port of a stats endpoint.
+        /// </param>
+        /// <remarks>
+        /// If the host or port are set to <c>null</c>/<c>0</c> this handler will be deactivated (every operation is a no-op).
+        /// Otherwise it sends data to a StatsD endpoint via UDP.
+        /// </remarks>
+        public StatsdMetricHandler(string host, ushort port)
+        {
+            Host = host;
+            Port = port;
+        }
+
+        /// <summary>
+        /// Host to which we should send metrics to.
+        /// </summary>
+        public string Host
+        {
+            get => _host;
+            set
+            {
+                _host = value;
+                var oldHandler = _activeHandler;
+                if (oldHandler != null)
+                {
+                    oldHandler.Dispose();
+                }
+
+                _activeHandler = GetHandler();
+            }
+        }
+
+        /// <summary>
+        /// Port to which we should send metrics to.
+        /// </summary>
+        public ushort Port
+        {
+            get => _port;
+            set
+            {
+                _port = value;
+                var oldHandler = _activeHandler;
+                if (oldHandler != null)
+                {
+                    oldHandler.Dispose();
+                }
+
+                _activeHandler = GetHandler();
+            }
+        }
+
+        /// <inheritdoc />
+        public IMetricReadingBatch BeginBatch() => _activeHandler.BeginBatch();
+
+        /// <inheritdoc />
+        public void SerializeMetadata(IEnumerable<Metadata> metadata) => _activeHandler.SerializeMetadata(metadata);
+
+        /// <inheritdoc />
+        public void SerializeMetric(in MetricReading reading) => _activeHandler.SerializeMetric(reading);
+
+        /// <inheritdoc />
+        public ValueTask FlushAsync(TimeSpan delayBetweenRetries, int maxRetries, Action<AfterSendInfo> afterSend,
+            Action<Exception> exceptionHandler) =>
+            _activeHandler.FlushAsync(delayBetweenRetries, maxRetries, afterSend, exceptionHandler);
+
+        /// <inheritdoc />
+        public void Dispose() => _activeHandler.Dispose();
+
+        private IMetricHandler GetHandler()
+        {
+            if (_host == null || _port == 0)
+            {
+                return NoOpMetricHandler.Instance;
+            }
+
+            return new BufferedStatsdMetricHandler(_host, _port);
+        }
+    }
+
     /// <summary>
     /// Implements <see cref="BufferedMetricHandler" /> by sending data to a statsd endpoint.
     /// </summary>
-    public class StatsdMetricHandler : BufferedMetricHandler
+    internal class BufferedStatsdMetricHandler : BufferedMetricHandler
```

Comment:
This can be `private` I think

Reply:
`SignalFxMetricHandler` will need to instantiate this one, too, that's why I couldn't make it `private`.

Reply:
Ah, good call, I missed that 👍